graylog2-server/src/main/java/org/graylog/plugins/views/search/permissions/SearchUser.java

```diff
@@ -53,6 +62,21 @@ public String username() {
 
     public boolean canReadView(ViewLike view) {
         final String viewId = view.id();
+
+        // If a resolved view id is provided, delegate the permissions check to the resolver.
+        final ViewResolverDecoder decoder = new ViewResolverDecoder(viewId);
+        if (decoder.isResolverViewId()) {
+            final ViewResolver viewResolver = viewResolvers.get(decoder.getResolverName());
+            if (viewResolver != null) {
+                return viewResolver.canReadView(viewId, isPermitted, isPermittedEntity);
+            } else {
+                // Resolved view could not be found, so permissions cannot be checked.
+                LOG.info("View resolver [{}] could not be found.", decoder.getResolverName());
+                return false;
+            }
+        }
```

Comment:
This implementation adds view resolver specific resolution into the `SearchUser` class. 

The positive: It's centralized, and the same permissions checks will be used everywhere uniformly (whether or not permissions are checked for a database view or a resolved view).

The negative: The conditional view resolver logic feels slightly out of place here. Are the concerns of DB-persisted views and resolved views combined too much here? I could not find a way to separate them. The only alternative I could think of, would be to perform the resolved view permissions checks completely separately from those of the DB views (like up in the resource classes). But that brings its own issues (for one, externalizing/distributing permissions checks into rest logic did not seem like an improvement). Feedback and alternate suggestions welcome.